packages/shared/components/inputs/Amount.svelte

```diff
@@ -25,32 +27,54 @@
     export let disabled = false
     export let autofocus = false
 
-    const Units = Object.values(Unit).filter((x) => x !== 'Pi')
+    let profileCurrency: AvailableExchangeRates = $activeProfile?.settings.currency ?? AvailableExchangeRates.USD
+
+    /** NOTE: Units is used for the dropdown, so it must contain both IOTA denominations and the user's selected fiat currency */
+    const Units = [profileCurrency as string].concat(Object.values(Unit).filter(x => x !== 'Pi').map(x => x as string))
     const MAX_VALUE = 2779530283000000
 
     let dropdown = false
     let navContainer
     let unitsButton
     let focusedItem
 
-    let profileCurrency: AvailableExchangeRates = $activeProfile?.settings.currency ?? AvailableExchangeRates.USD
-    $: fiatAmount = amountToFiat(amount)
+    $: amountForLabel = getFormattedLabel(amount)
     $: {
         if (amount.length > 0) {
-            const rawVal = changeUnits(parseCurrency(amount), unit, Unit.i)
-            if (rawVal > MAX_VALUE) {
-                amount = formatUnitPrecision(MAX_VALUE, unit, false)
+            if(!isFiatCurrency(unit)) {
+                const rawVal = changeUnits(parseCurrency(amount), unit, Unit.i)
+                if (rawVal > MAX_VALUE) {
+                    amount = formatUnitPrecision(MAX_VALUE, unit, false)
+                }
+            } else {
+                const amountAsI = convertFromFiat(amount, $currencies[CurrencyTypes.USD], $exchangeRates[profileCurrency])
+                if(amountAsI > MAX_VALUE) {
+                    amount = convertToFiat(MAX_VALUE, $currencies[CurrencyTypes.USD], $exchangeRates[profileCurrency]).toString()
+                }
             }
         }
     }
 
     const clickOutside = () => {
         dropdown = false
     }
+
     const onSelect = (index) => {
         if (amount.length > 0) {
-            amount = formatUnitPrecision(changeUnits(parseCurrency(amount), unit, Unit.i), index, false)
+            if(isFiatCurrency(index)) {
+                if(isFiatCurrency(unit)) return
```

Comment:
I am not sure why we need an early return here. Is there a situation where if the provided `index` is fiat, the unit will also be a `fiat`? IIUC, this line isn't needed.

Reply:
> I am not sure why we need an early return here. Is there a situation where if the provided `index` is fiat, the unit will also be a `fiat`? IIUC, this line isn't needed.

The early return statement covers the FIAT -> FIAT case, which only happens when the user decides to not change anything and just reselects the same fiat currency. If this check isn't made, then the fiat amount gets re-calculated into fiat again, which when I tried it just kept reducing the amount. I realize now though that this could be re-written just as `amount = isFiatCurrency(unit) ? amount : amountToFiat(amount).slice(2)`...

Reply:
Just tried re-writing some of it to help make it cleaner... how does this look? 

```
const onSelect = (index) => {
    updateUnit(index)
}

const updateUnit = (_unit) => {
    if(unit === _unit) return

    convertAmount(unit, _unit)

    unit = _unit
}

const convertAmount = (fromUnit, toUnit) => {
    if(amount.length <= 0) return

    if(isFiatCurrency(toUnit)) {
        amount = amountToFiat(amount).slice(2)
    } else {
        if(isFiatCurrency(fromUnit)) {
            let amountAsI = convertFromFiat(amount, $currencies[CurrencyTypes.USD], $exchangeRates[profileCurrency])
            amount = formatUnitPrecision(amountAsI, toUnit).replace(toUnit, '')
        } else {
            amount = formatUnitPrecision(changeUnits(parseCurrency(amount), fromUnit, Unit.i), toUnit, false)
        }
    }
}
```

Reply:
@maxwellmattryan This is a lot cleaner. Looks good, thank you for the revision.